tracks/java/exercises/isogram/mentoring.md

```diff
@@ -0,0 +1,40 @@
+# Mentoring
+
+## Reasonable Solutions
+
+### Using Streams
+
+```java
+public class IsogramChecker {
+    public boolean isIsogram(String word) {
+
+        String noSpacesOrHyphens = word.replaceAll("(?:\\s|-)", "").toLowerCase();
+
+        return noSpacesOrHyphens.chars().distinct().count() == noSpacesOrHyphens.length();        
+    }
+}
+
+```
+
+### Using a Set
+
+```java
+class IsogramChecker {
+  public boolean isIsogram(String input) {
+    Set<Character> tmp = new HashSet<Character>();
+    for(char ch : input.toLowerCase().toCharArray()) {
+        if (Character.isLetter(ch) && !tmp.add(ch)) {
+            return false;
+        }
+    }
+    return true;
+  }
+}
+```
+
+## Talking points
+
+- Solutions that use the Collections Framework should opt for a Set interface to check for duplicates instead of Maps and Lists. Sets don’t allow duplicates, so they are the best fit for this check.
```

Comment:
```suggestion
- Solutions that use the Collections Framework should opt for a Set interface to check for duplicates instead of Maps and Lists. Sets do not allow duplicates, so they are the best fit for this check.
```